### DISCUSSION TITLE
SQL Database Connection

### DISCUSSION THREAD

**Discussion by insomniacj93** [Budibase/budibase#6846] [Help] [answered]
Hello,

Since Update 1.1.2/4 that i've been experiencing alot of issues with the database connections, before this, any changes that were done to the MySQL tables (new columns, tables, rows created, delete certain columns or rows, etc.) all this sort of changes would be noticeable inside the database interface in budibase. Right now the tables won't update themselves, it requires me to fetch them, once fetched, then the rows and columns get stuck, not allowing me to delete them. It's very messy. The solution i've found is to basicly Delete the database connection and create a new one. Which is not ideal as you might believe, that implies re-connecting everysingle component in the applications.

I'm currently working with the latest 1.1.17 with docker-compose.

Any help will be very much appreciated!

**Comment by melohagan** [comment]
Hello @insomniacj93 

This user may also be having a similar issue: https://github.com/Budibase/budibase/issues/6653

If you check the console/network tabs (Fn+F12 on Chrome) do you see any error messages when the data is not updating as you expect?

**Reply by insomniacj93** [reply]
Caught it, the column "condition_check" no longer exists.
But even though budibase notices that it is no longer there, in builder we can still see it. 
After trying to delete the column I get the following errors (the last 2).

![print6](https://user-images.githubusercontent.com/107686004/180778129-7e14db5f-1d03-4f63-9962-090cd43d369b.png)

Thank you in advance @melohagan !

**Reply by melohagan** [reply]
I think this is related to this issue here, which I believe you've had a look at: https://github.com/Budibase/budibase/issues/6737

Also out of curiosity, what's inside the error object if you click on the arrow to expand. (The second error in that screenshot).

**Reply by mmalessa80** [reply]
![image](https://user-images.githubusercontent.com/106758851/180880705-369ab54e-de78-4027-8cdd-35f3cda035a5.png)

Here is one of my errors expanded.

**Reply by insomniacj93** [reply]
@melohagan 
Sorry, Only saw your reply today.

![print7](https://user-images.githubusercontent.com/107686004/180949148-57a3d415-c8dd-41cc-93b3-68127274d881.png)

It does recognise the missing column, thing is, it is still there.

**Comment by mmalessa80** [comment]
@melohagan I had briefly mentioned this issue here https://github.com/Budibase/budibase/discussions/6651. I had previously dropped the ID column from this table via Cloud Shell using the drop column command. Then in Budibase, I fetched the DB tables again, however the column still shows within Budibase and even if I delete it in Budibase it still remains there. Also, now my table rows don't load because I think Budibase still thinks the ID column is the primary key, although the column no longer exists. Currently the only plausible solution might be to drop the datasource and then add it again. However, this would result in losing the entire data structure including relationships and bindings to the UI design, which does not seem like a reasonable solution.

![image](https://user-images.githubusercontent.com/106758851/180880003-a0d8a3d2-6996-4270-9a9d-bb974091fc86.png)

**Reply by insomniacj93** [reply]
@mmalessa80 feels like we are having the same issue, I will let you know if i found a fix!

**Comment by mmalessa80** [comment]
Just wondering if there’s been any progress/resolution on this?

**Reply by insomniacj93** [reply]
@mmalessa80 I'm afraid not..

**Comment by mike12345567** [comment, marked as answer]
Hi @mmalessa80 and @insomniacj93 - A fix for this issue will be released with the next release (v. soon)..

**Reply by insomniacj93** [reply]
Hi, thanks @mike12345567 !!
